src/bindings/python/src/openvino/_op_base.py

```diff
@@ -13,9 +13,22 @@ def __init__(self, py_obj: "Op", inputs: Optional[Union[List[Union[Node, Output]
         super().__init__(py_obj)
         self._update_type_info()
         if isinstance(inputs, tuple):
-            inputs = None if len(inputs) == 0 else list(inputs)
+            # inputs = None if len(inputs) == 0 else list(inputs)
+            if len(inputs) == 0:
+                inputs = None
             if inputs is not None and len(inputs) == 1 and isinstance(inputs[0], list):
                 inputs = inputs[0]
+
+        if isinstance(inputs,list):
+            flat_list = []
+            for items in inputs:
+                if isinstance(items,list):
+                    flat_list.extend(items)
+                else:
+                    flat_list.append(items)
+            inputs = flat_list
+
         if inputs is not None:
-            self.set_arguments(inputs)
+            output_list = [item for item in inputs if isinstance(item, Output)]
+            self.set_arguments(output_list)
```

Comment:
actually `set_arguments` may be both list of Nodes or Outputs [link](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/src/pyopenvino/graph/node.cpp#L511)